data-prepper-plugins/mutate-string-processors/src/main/java/com/amazon/dataprepper/plugins/processor/mutatestring/SplitStringProcessor.java

```diff
@@ -25,14 +26,45 @@ public SplitStringProcessor(final PluginMetrics pluginMetrics, final SplitString
         super(pluginMetrics, config);
 
         for (SplitStringProcessorConfig.Entry entry: config.getEntries()) {
-            patternMap.put(entry.getDelimiter(), Pattern.compile(entry.getDelimiter()));
+            if(entry.getDelimiterRegex() != null && !entry.getDelimiterRegex().isEmpty()
+                    && entry.getDelimiter() != null && !entry.getDelimiter().isEmpty()) {
+                throw new IllegalArgumentException("delimiter and delimiter_regex cannot be defined at the same time");
+            } else if((entry.getDelimiterRegex() == null || entry.getDelimiterRegex().isEmpty()) &&
+                    (entry.getDelimiter() == null || entry.getDelimiter().isEmpty())) {
+                throw new IllegalArgumentException("delimiter or delimiter_regex needs to be defined");
+            }
+
+            if(entry.getDelimiterRegex() != null && !entry.getDelimiterRegex().isEmpty()) {
+                patternMap.put(entry.getDelimiterRegex(), Pattern.compile(entry.getDelimiterRegex()));
+            } else {
+                if(isRegexChar(entry.getDelimiter())) {
```

Comment:
I believe you can use [Pattern::quote](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html#quote(java.lang.String)) to handle all of this code.

`Pattern.compile(Pattern.quote(entry.getDelimiterRegex()))`

Reply:
Oh, nice!